rust/feature-flags/src/flag_matching.rs

```diff
@@ -1105,15 +1116,69 @@ impl FeatureFlagMatcher {
     }
 }
 
+/// This function checks if a person is in a static cohort by querying the `posthog_cohortpeople` table.
+async fn evaluate_static_cohort_filter(
+    postgres_reader: PostgresReader,
+    team_id: TeamId,
+    distinct_id: String,
+    cohort_id: CohortId,
+) -> Result<bool, FlagError> {
+    let mut conn = postgres_reader.get_connection().await?;
+
+    let query = r#"
+            SELECT EXISTS (
+                SELECT 1
+                FROM posthog_cohortpeople
```

Comment:
future improvement, this ought to query all static cohorts inclusion/exclusion at once

Reply:
and if you already got person properties, you should have the person_id as well, so you can skip the person_id fetching subquery

Reply:
ah Olly said the same below 🙈